jax/lax/lax.py

```diff
@@ -3341,7 +3341,7 @@ def _scatter_jvp(primals, tangents, update_jaxpr, update_consts,
   new_operand = pad(new_operand, _zero(operand),
                     ((0, 1, 0),) + tuple((0, 0, 0) for _ in operand_shape))
 
-  ids_shape = onp.array(updates_shape)
+  ids_shape = onp.array(updates_shape, dtype=onp.int32)
```

Comment:
Shouldn't `update_shape` always be a tuple of Python integers? It feels a little fishy that we need this, I wonder if there's another latent bug somewhere.

Reply:
I'm not sure. Paging @mattjj

Reply:
It looks like this can come up because `updates_shape` can be an empty tuple, and NumPy defaults to float64 for empty tuples. Maybe just add a comment noting why we need a dtype?

Reply:
Added a comment.